theseus/embodied/robot/joint.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+import warnings
+from typing import Optional
+import torch
+
+from theseus.geometry.functional import so3, se3
+
+
+class Joint(abc.ABC):
+    def __init__(
+        self,
+        name: str,
+        parent: int = -1,
+        child: int = -1,
+        origin: Optional[torch.Tensor] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        if origin is None and dtype is None:
+            dtype = torch.get_default_dtype()
+        if origin is not None:
+            if origin.shape[0] != 1 or not se3.check_group_tensor(origin):
+                raise ValueError("Origin must be an element of SE(3).")
+
+            if dtype is not None and origin.dtype != dtype:
+                warnings.warn(
+                    f"tensor.dtype {origin.dtype} does not match given dtype {dtype}, "
+                    "tensor.dtype will take precendence."
+                )
+            dtype = origin.dtype
+        else:
+            origin = torch.zeros(1, 3, 4, dtype=dtype)
+            origin[:, 0, 0] = 1
+            origin[:, 1, 1] = 1
+            origin[:, 2, 2] = 1
```

Comment:
Could we add a identity SE3 helper function and then call that here?

Reply:
Will go back to this once `identity()` is added to the Lie group library.